frontend/ui/impl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/UnmanagedBranchNotificationFactory.java

```diff
@@ -0,0 +1,117 @@
+
+  private final Project project;
+  private final @Nullable IGitMacheteRepositorySnapshot gitMacheteRepositorySnapshot;
+  private final String branchName;
+  private final String inferredParent;
+
+  public Notification create() {
+    val notification = VcsNotifier.STANDARD_NOTIFICATION.createNotification(
+        getString("action.GitMachete.EnhancedGraphTable.unmanaged-branch-notification.text").format(branchName),
+        NotificationType.INFORMATION);
+
+    val slideInAction = getSlideInAction(notification);
+    val dontShowForThisBranchAction = getDontShowForThisBranchAction(notification);
+    val dontShowForThisProjectAction = getDontShowForThisProjectAction(notification);
+
+    notification.addAction(slideInAction);
+    notification.addAction(dontShowForThisBranchAction);
+    notification.addAction(dontShowForThisProjectAction);
+
+    return notification;
+  }
+
+  @UIEffect
+  public static boolean showForThisProject() {
+    return PropertiesComponent.getInstance().getBoolean(SHOW_UNMANAGED_BRANCH_NOTIFICATION, /* defaultValue */ true);
+  }
+
+  @UIEffect
+  public static boolean showForThisBranch(String aBranchName) {
+    return PropertiesComponent.getInstance().getBoolean("${SHOW_UNMANAGED_BRANCH_NOTIFICATION}.${aBranchName}",
+        /* defaultValue */ true);
+  }
+
+  private NotificationAction getSlideInAction(Notification notification) {
+    return NotificationAction
+        .createSimple(
+            getString("action.GitMachete.EnhancedGraphTable.unmanaged-branch-notification.action.slide-in")
+                .format(inferredParent),
+            () -> {
+              val dataContext = new DataContext() {
+                @Override
+                public @Nullable Object getData(String dataId) {
+                  return Match(dataId).of(
+                      typeSafeCase(DataKeys.GIT_MACHETE_REPOSITORY_SNAPSHOT, gitMacheteRepositorySnapshot),
+                      typeSafeCase(DataKeys.SELECTED_BRANCH_NAME, inferredParent),
+                      typeSafeCase(DataKeys.UNMANAGED_BRANCH_NAME, branchName),
+                      typeSafeCase(CommonDataKeys.PROJECT, project),
+                      Case($(), (Object) null));
+                }
+              };
+              val actionEvent = AnActionEvent.createFromDataContext(ActionPlaces.VCS_NOTIFICATION, new Presentation(),
+                  dataContext);
+              ActionManager.getInstance().getAction(SLIDE_IN_UNMANAGED_BELOW).actionPerformed(actionEvent);
+              notification.expire();
+            });
+  }
+
+  private NotificationAction getDontShowForThisBranchAction(Notification notification) {
+    return NotificationAction
+        .createSimple(
+            getString("action.GitMachete.EnhancedGraphTable.unmanaged-branch-notification.action.dont-show-for-branch")
+                .format(branchName),
+            () -> {
+              String propertyKey = "${SHOW_UNMANAGED_BRANCH_NOTIFICATION}.${branchName}";
```

Comment:
I could not find where are these props stored 😕 on the other hand Ive tested the feature and it seems to work correctly

Reply:
I found it here
`/Users/mkondratek/IdeaProjects/git-machete-intellij-plugin/build/idea-sandbox/config/options/other.xml`
❗